pkg/controller/defaultdomaincert/default_domain_cert_controller_test.go

```diff
@@ -0,0 +1,875 @@
+package defaultdomaincert
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	approutingv1alpha1 "github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	"github.com/Azure/aks-app-routing-operator/pkg/config"
+	"github.com/Azure/aks-app-routing-operator/pkg/controller/testutils"
+	"github.com/Azure/aks-app-routing-operator/pkg/manifests"
+	"github.com/Azure/aks-app-routing-operator/pkg/store"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// ErrorClient wraps a client to inject errors for testing
+type ErrorClient struct {
+	client.Client
+	GetError    error
+	CreateError error
+	UpdateError error
+	DeleteError error
+	PatchError  error
+}
+
+func (e *ErrorClient) Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+	if e.GetError != nil {
+		return e.GetError
+	}
+	return e.Client.Get(ctx, key, obj, opts...)
+}
+
+func (e *ErrorClient) Create(ctx context.Context, obj client.Object, opts ...client.CreateOption) error {
+	if e.CreateError != nil {
+		return e.CreateError
+	}
+	return e.Client.Create(ctx, obj, opts...)
+}
+
+func (e *ErrorClient) Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
+	if e.UpdateError != nil {
+		return e.UpdateError
+	}
+	return e.Client.Update(ctx, obj, opts...)
+}
+
+func (e *ErrorClient) Patch(ctx context.Context, obj client.Object, patch client.Patch, opts ...client.PatchOption) error {
+	if e.PatchError != nil {
+		return e.PatchError
+	}
+	return e.Client.Patch(ctx, obj, patch, opts...)
+}
+
+func (e *ErrorClient) Delete(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error {
+	if e.DeleteError != nil {
+		return e.DeleteError
+	}
+	return e.Client.Delete(ctx, obj, opts...)
+}
+
+const (
+	testNamespace   = "test-namespace"
+	testSecretName  = "test-secret"
+	testCertPath    = "/path/to/cert.crt"
+	testKeyPath     = "/path/to/key.key"
+	testCertContent = "-----BEGIN CERTIFICATE-----\nMIIC...test cert content...\n-----END CERTIFICATE-----"
+	testKeyContent  = "-----BEGIN PRIVATE KEY-----\nMIIE...test key content...\n-----END PRIVATE KEY-----"
+)
+
+// mockStore implements the store.Store interface for testing
+type mockStore struct {
```

Comment:
wonder if this is better included in store pkg?

Reply:
it's fine as is, we only use this here